homeassistant/components/tolo/config_flow.py

```diff
@@ -37,29 +49,48 @@ def _check_device_availability(host: str) -> bool:
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
-        """Handle a flow initialized by the user."""
+        """Handle a config flow initialized by the user."""
         errors = {}
 
         if user_input is not None:
-            self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
-
             device_available = await self.hass.async_add_executor_job(
                 self._check_device_availability, user_input[CONF_HOST]
             )
 
-            if not device_available:
-                errors["base"] = "cannot_connect"
-            else:
-                return self.async_create_entry(
-                    title=DEFAULT_NAME, data={CONF_HOST: user_input[CONF_HOST]}
-                )
+            if device_available:
+                if self.source == SOURCE_RECONFIGURE:
```

Comment:
you'll need to check here if the unqiue id has changed, to ensure a user doesn't reconfigure to a different device

Reply:
How to do that?

My Problem: This device does not necessarily have a unique id. Additionally, I thought that `self._get_reconfigure_entry()` just provides me with the correct config entry, regardless if it has a unique id or not - and updates exactly this config entry again. Am I wrong here?

Reply:
> Additionally, I thought that self._get_reconfigure_entry() just provides me with the correct config entry, regardless if it has a unique id or not -

Correct, the issue is not checking if Home Assistant is providing the same device, but if the user re-auths against the same device as the config entry that is being re-authed.

Example, considering I have two Spotify accounts in Home Assistant. Mine & that one of my wife. Now, if I reauth mine, but login to my wife's account... that is a mismatch. That is what is meant by Josef in the comment above.

Reply:
I believe we are on the same track. It's actually a bit strange tbh that a unique id is only set on discovery as this means a user could add a discovered device again. Is there nothing we can get from the `ToloClient` which helps us idenfity the current device e.g. serial no, identifier, MAC?

Reply:
I wrote the `tololib` package myself. A Tolo device communicates via a TCP/UDP to serial converter. This is what introduces a mac address - but the UDP communication is transparently forwarded. So, no way to get a MAC address add this point.

Yes, I could scrape the web interface of the converter, but it is not ensured that the login credentials/ports are the same everywhere (Actually, I had hand on two devices with different settings here). The only common is the UDP port and protocol used for the actual Tolo communication - transparently.

The protocol itself is quite basic, there is no documentation and I got it by reverse-engineering the (never published original) Android app of Tolo, which got later confirmed by the manufacturer to be correct. But no, there is no ID or serial on that device itself.

Reply:
Any news on this? How can we proceed here? I really would like to get that merged...

Reply:
Could you please give me some advice how/where to do that? Just calling `_async_abort_entries_match` I guess will not work as, when I'm not changing the IP address, the entry is still there, correct?

Reply:
you are checking `_async_abort_entries_match(CONF_IP: new_IP)` so it should only abort if you already have an entry with the new IP already, which is exactly what we want

Reply:
Ok, not sure if I get you right now... so everything you want is already there? If so, can we then merge this?

Reply:
```suggestion
                if self.source == SOURCE_RECONFIGURE:
                    self._abort_entries_match(CONF_HOST: user_input[CONF_HOST])
```
you're not checking it yet are you?

Reply:
I think I am, thanks, and sorry for not knowing this.

One last question: Does `_abort_entries_match` automatically recognize the "current" vs. "other" config entries? If not, that would abort if I do *not* change the IP address, or am I wrong?

Reply:
No worries. It would abort, but you only have this one parameter in your schema, meaning that the user didn't change anything. You can check `user_input[CONF_HOST] != self._get_reconfigure_entry().data[CONF_HOST]` and just continue otherwise